envoy/api/v2/core/health_check.proto

```diff
@@ -153,6 +153,28 @@ message HealthCheck {
   //
   // The default value for "no traffic interval" is 60 seconds.
   google.protobuf.Duration no_traffic_interval = 12;
+
+  // The "unhealthy interval" is a health check interval that is used for hosts that are marked as
+  // unhealthy. As soon as the host is marked as healthy, Envoy will shift back to using the
+  // standard health check interval that is defined.
+  //
+  // The default value for "unhealthy interval" is the same as "interval".
+  google.protobuf.Duration unhealthy_interval = 14;
+
+  // The "unhealthy edge interval" is a special health check interval that is used for the first
+  // health check right after a host is marked as unhealthy. For subsequent health checks
+  // Envoy will shift back to using either "unhealthy interval" if present or the standard health
+  // check interval that is defined.
+  //
+  // The default value for "unhealthy edge interval" is the same as "unhealthy interval".
+  google.protobuf.Duration unhealthy_edge_interval = 15;
```

Comment:
Mind documenting that no_traffic_interval_ takes precedence on these new fields, even if it's qualifiying "used for traffic-bearing hosts" or something

Reply:
alternately this can be done later if Matt wants to merge this in before we swap repos around :-)

Reply:
Done. @alyssawilk PTAL at the change to `no_traffic_interval` docs.